stonesoup/updater/kalman.py

```diff
@@ -901,3 +902,132 @@ def predict_measurement(self, predicted_state, measurement_model=None, measureme
 
         return MeasurementPrediction.from_state(
             predicted_state, meas_pred_mean, meas_pred_covar, cross_covar=cross_covar)
+
+
+class StochasticIntegrationUpdater(KalmanUpdater):
+    """Stochastic Integration Kalman Filter class. Inherits most
+    of the functionality from :class:`~.KalmanUpdater`.
+
+    The measurement update of nonlinear measurement models is accomplished by
+    the stochastic integration approximation.
+
+    """
+
+    # Can be non-linear and non-differentiable
+    measurement_model: MeasurementModel = Property(
+        default=None,
+        doc="The measurement model to be used. This need not be defined if a "
+        "measurement model is provided in the measurement. If no model "
+        "specified on construction, or in the measurement, then error "
+        "will be thrown.",
+    )
+    Nmax: float = Property(default=10, doc="maximal number of iterations of SIR")
+    Nmin: float = Property(
+        default=5,
+        doc="minimal number of iterations of stochastic integration rule (SIR)",
+    )
+    Eps: float = Property(default=5e-4, doc="allowed threshold for integration error")
+    SIorder: float = Property(
+        default=5, doc="order of SIR (orders 1, 3, 5 are currently supported)"
+    )
+
+    @lru_cache()
+    def predict_measurement(
+        self, predicted_state, measurement_model=None, measurement_noise=True, **kwargs
+    ):
+        """SIF.
```

Comment:
Expand on what is done in this function, similar to other updaters. One line will be sufficient!

Reply:
Added in de02cd5fd7780094396e6b53e4294b08a2259b4d